agent/worker.go

```diff
@@ -568,9 +573,9 @@ func (w *worker) Subscribe(ctx context.Context, subscription *api.SubscriptionMe
 		case v := <-ch:
 			task := v.(*api.Task)
 			if match(task) {
-				w.mu.Lock()
+				w.mu.RLock()
```

Comment:
The statement is evaluated, then the primary function call is dispatched. So, the map lookup happens before it heads to the go routine.